joomla/joomla-cms
# [4.0] Edit UpdateSite - GSoC Expand Extensions Manager #17553 Redo

This is a continuation of issue #17533 as I can no longer push to that repo, starting a new PR.

### Introduction

In the past, the Update Site was a hidden location, only managed by the extension. Nowadays, users can Publish/Unpublish, Delete an Update Site. Following the trend to increase the user control and security over the Update Site location, the option to edit the URL will enable the following use changes:

- To confirm the address where the updates are coming from
- Change the address where the updates are coming from
- In the future, the edition of the Update Site will allow to further enrich the information provided as part of an Update Site (e.g. RSA Public Keys).
- As preparation for handling signed update packages, where public keys may need to be added manually, a first step is to make the update sites records editable.
- This can also be useful if an extension developer has changed the update URL. Users can still update after changing the update location.

### Summary of Changes

Enhanced Update Site

The new feature will allow the user to edit the update site URL:

<img width="514" alt="image" src="https://user-images.githubusercontent.com/359377/54067741-425b4d80-4244-11e9-9a67-10c35535928b.png">


### Testing Instructions

1. Apply the patch
2. Go to System
3. Click on Update Sources
<img width="299" alt="image" src="https://user-images.githubusercontent.com/359377/54067774-9bc37c80-4244-11e9-9dda-e1c591d39ec1.png">

4. Click on the name of an update site to edit it or select the checkbox and click Edit in the toolbar
5. Change the location URL
6. Click Save or Save & Close
7. Notice the update site is changed

You can select any update site by clicking in its name or selecting the checkbox and then Edit.

### Expected result

<img width="609" alt="image" src="https://user-images.githubusercontent.com/359377/54067801-070d4e80-4245-11e9-9299-929add174906.png">


### Actual result

<img width="625" alt="image" src="https://user-images.githubusercontent.com/359377/54067807-17bdc480-4245-11e9-89c1-4bb8042e97b2.png">
